一定要设置CLOUDFLARE_API_TOKEN 吗？

<img width="2888" height="488" alt="Image" src="https://github.com/user-attachments/assets/3096f92c-d511-41e0-98e5-f586e4cf59f6" />

[ERROR] In a non-interactive environment, it's necessary to set a CLOUDFLARE_API_TOKEN environment variable for wrangler to work. Please go to https://developers.cloudflare.com/fundamentals/api/get-started/create-token/ for instructions on how to create an api token, and assign its value to CLOUDFLARE_API_TOKEN.

--- a/README.md
+++ b/README.md
@@ -23,7 +23,7 @@ npm run deploy
 
 **Prerequisites:**
 - [NVIDIA API Key](https://build.nvidia.com) - Required during `npm run setup`
-- Cloudflare account - Will prompt to login during first `npm run deploy`
+- Cloudflare account - `npm run setup` will prompt to login on first run (or set `CLOUDFLARE_API_TOKEN` env var for non-interactive use)
 
 After deployment, you'll get a Worker URL: `https://nvidia-anthropic-proxy.xxx.workers.dev`
 

--- a/README_CN.md
+++ b/README_CN.md
@@ -23,7 +23,7 @@ npm run deploy
 
 **准备工作：**
 - [NVIDIA API Key](https://build.nvidia.com) - 在 `npm run setup` 时输入
-- Cloudflare 账户 - 首次 `npm run deploy` 时会提示登录
+- Cloudflare 账户 - 首次 `npm run setup` 时会提示登录（或设置 `CLOUDFLARE_API_TOKEN` 环境变量以非交互方式使用）
 
 部署成功后会显示 Worker 地址：`https://nvidia-anthropic-proxy.xxx.workers.dev`
 

--- a/setup.sh
+++ b/setup.sh
@@ -15,6 +15,14 @@ npm install
 
 echo
 
+# 确保已登录 Cloudflare：pipe 给 wrangler 会让 stdin 变成非交互式，
+# 此时若未登录就会强制要 CLOUDFLARE_API_TOKEN，所以先在这里登录
+if ! npx wrangler whoami >/dev/null 2>&1; then
+    echo "Cloudflare not logged in. Opening browser to login..."
+    npx wrangler login
+    echo
+fi
+
 # 配置 NVIDIA_API_KEY
 echo "Enter your NVIDIA API Key:"
 read -s NVIDIA_KEY

